Cargo.toml

```diff
@@ -153,6 +153,7 @@ hex = { version = "0.4.3" }
 indexmap = "2.10.0"
 itertools = "0.14"
 log = "^0.4"
+lru = "0.16.0"
```

Comment:
The [`lru`](https://crates.io/crates/lru) crate appears to be well maintained and used by a number of popular crates (like `tracing-log`, `redis`, and `aws-sdk-s3`). Let me know if it is ok to include it.

Reply:
I looked at the (very nicely) coded lru in this PR and I think it is quite good. Unless there are any objections, I think we can go with what is here